CouchbaseBucketGoCB.GetAndTouchRaw() passes incorrect return param to gocb.GetAndTouch()

I came across this during https://github.com/couchbase/sync_gateway/issues/2561 development.

Steps to repro:

1. Switch to https://github.com/couchbase/sync_gateway/issues/2561 feature branch 
1. `export SG_TEST_BACKING_STORE=Couchbase`
1. `go test -v -run '^TestGetAndTouchRaw$' github.com/couchbase/sync_gateway/base`

Output:

```
=== RUN   TestGetAndTouchRaw
2017-05-15T16:49:39.732-07:00 GoCBGoCouchbaseHybrid Opening Couchbase database testbucket on <http://localhost:8091> as user "testbucket"
_time=2017-05-15T16:49:39.740-07:00 _level=INFO _msg= Trying with http://127.0.0.1:8091/pools/default/bucketsStreaming/testbucket
2017/05/15 16:49:39 Error calling GetAndTouchRaw: invalid character 'b' looking for beginning of value

	bucket_gocb_test.go:737
	assert.True(t, err == nil)

	bucket_gocb_test.go:738
	assert.Equals(t, len(rv), len(val))

	expected: 3
	     got: 0
--- FAIL: TestGetAndTouchRaw (0.15s)
	assert.go:31:
	assert.go:31:
FAIL
exit status 1
FAIL	github.com/couchbase/sync_gateway/base	0.161s
```

--- a/base/bucket_gocb.go
+++ b/base/bucket_gocb.go
@@ -659,7 +659,7 @@ func (bucket CouchbaseBucketGoCB) GetAndTouchRaw(k string, exp int) (rv []byte, 
 		gocbExpvars.Add("SingleOps", -1)
 	}()
 
-	var returnVal interface{}
+	var returnVal []byte
 	worker := func() (shouldRetry bool, err error, value interface{}) {
 
 		gocbExpvars.Add("GetAndTouchRaw", 1)
@@ -689,7 +689,7 @@ func (bucket CouchbaseBucketGoCB) GetAndTouchRaw(k string, exp int) (rv []byte, 
 		return nil, cas, err
 	}
 
-	return returnVal.([]byte), cas, err
+	return returnVal, cas, err
 
 }
 

